oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/RelatedSelectField/RelatedSelectField.jsx

```diff
@@ -0,0 +1,250 @@
+    // maybe it makes sense for the component to accept searchConfig as a prop
+
+    const searchConfig = {
+      searchApi: {
+        axios: {
+          headers: suggestionAPIHeaders,
+          url: externalSuggestionAPI,
+          withCredentials: false,
+        },
+      },
+      initialQueryState: {
+        queryString: this.state.searchQuery,
+        size: 10,
+      },
+      paginationOptions: {
+        defaultValue: 10,
+        resultsPerPage: [
+          { text: "10", value: 10 },
+          { text: "20", value: 20 },
+          { text: "50", value: 50 },
+        ],
+      },
+      layoutOptions: {
+        listView: true,
+        gridView: false,
+      },
+    };
+    return (
+      <React.Fragment>
+        {super.render(this.getProps())}
+        {externalSuggestionAPI && (
+          <ExternalApiModal
+            searchConfig={searchConfig}
+            open={this.state.isModalOpen}
+            onClose={this.handleModal}
+            serializeSuggestions={
+              serializeExternalApiSuggestions !== undefined
+                ? serializeExternalApiSuggestions
+                : serializeSuggestions
+            }
+            handleAddingExternalApiSuggestion={
+              this.handleAddingExternalApiSuggestion
+            }
+            fieldPath={fieldPath}
+          />
+        )}
+      </React.Fragment>
+    );
+  }
+}
+
+RelatedSelectField.propTypes = {
+  fieldPath: PropTypes.string.isRequired,
+  suggestionAPIUrl: PropTypes.string.isRequired,
+  suggestionAPIQueryParams: PropTypes.object,
+  suggestionAPIHeaders: PropTypes.object,
+  serializeSuggestions: PropTypes.func,
+  serializeAddedValue: PropTypes.func,
+  initialSuggestions: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.object),
+    PropTypes.object,
+  ]),
+  debounceTime: PropTypes.number,
+  noResultsMessage: PropTypes.node,
+  loadingMessage: PropTypes.string,
+  suggestionsErrorMessage: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.object,
+  ]),
+  noQueryMessage: PropTypes.string,
+  preSearchChange: PropTypes.func, // Takes a string and returns a string
+  onValueChange: PropTypes.func, // Takes the SUI hanf and updated selectedSuggestions
+  search: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
+  multiple: PropTypes.bool,
+  externalSuggestionAPI: PropTypes.string,
+  serializeExternalApiSuggestions: PropTypes.func,
+};
+
+RelatedSelectField.defaultProps = {
+  debounceTime: 500,
+  suggestionAPIQueryParams: {},
+  serializeSuggestions: serializeSuggestions,
+  suggestionsErrorMessage: "Something went wrong...",
+  noQueryMessage: "Search...",
+  noResultsMessage: "No results found.",
+  loadingMessage: "Loading...",
+  preSearchChange: (x) => x,
+  search: true,
+  multiple: false,
+  serializeAddedValue: undefined,
+  initialSuggestions: [],
+  onValueChange: undefined,
+  suggestionAPIHeaders: { Accept: "application/json" },
+  serializeExternalApiSuggestions: undefined,
```

Comment:
It is already getting the default serializer via props, though I do see that the naming was not good, so probably it was unclear because of that. I changed it like this

     <ExternalApiModal
            searchConfig={searchConfig}
            open={this.state.isModalOpen}
            onClose={this.handleModal}
            **serializeExternalApiSuggestions={
              serializeExternalApiSuggestions
                ? serializeExternalApiSuggestions
                : serializeSuggestions
            }**
            handleAddingExternalApiSuggestion={
              this.handleAddingExternalApiSuggestion
            }
            fieldPath={fieldPath}
          />
        )}